go-slang/examples/hof.go

```diff
@@ -0,0 +1,15 @@
+package main
+
+func add(a int, b int) int {
+  return a + b
+}
+
+func applyFunc(f func(int, int) int) int {
+  x := f(3, 5)
+  return x
+}
+
+func main() {
+  result := applyFunc(add)
+  Println(result)
+}
```

Comment:
can we put both types of HOF in the example? feel like he might ask and it's good to have both on hand

Reply:
updated!